app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUpdate.kt

```diff
@@ -28,13 +28,17 @@ class CustomPrescriptionEntryUpdate : Update<CustomPrescriptionEntryModel, Custo
       model: CustomPrescriptionEntryModel,
       prescription: PrescribedDrug
   ): Next<CustomPrescriptionEntryModel, CustomPrescriptionEntryEffect> {
-    var updatedModel = model
-        .drugNameChanged(prescription.name)
+    return if (prescription.isDeleted) {
```

Comment:
Ugh, it took me a moment to realise how this works because the deletion is happening elsewhere and we are depending on a reactive change notification to get this.

Ideally, I'd like to change this to make the deletion of the prescribed drug explicit, i.e., forward an event from the UI to the loop once the deletion is done and change loading of prescribed drug to an synchronous call instead of a reactive one.

What do you think? If this sounds good, we can add a comment for now explaining how this works and then add a ticket for fixing this flow.

Reply:
I will raise a ticket for this.